src-built-in/components/authentication/app.jsx

```diff
@@ -0,0 +1,93 @@
+    const { handleChange, formValues, proceedToAuthorize, quitApplication } = useAuth();
+    const { error, setError } = useState(null);
+
+    /*
+     * Replace this function with your own authentication method that makes the call to your authentication server.
+     */
+    const authenticateUser = async () => {
+        // The user form data to be sent to the authentication server.
+        const { username, password } = formValues;
+        // Replace this with your own authentication calls. Currently it will always authenticate no matter what the user input is.
+        const result = await Promise.resolve({res: "ok"});
+        // the result is an object containing a response and an error. For example:
+        // {
+        //     res: "ok",
+        //     error: "Password is not correct"
+        // }
+        return result;
+    }
+
+    /**
+     * This function invokes the function that makes the call to the server that verifies user authentication information and
+     * decides whether to authorize the user to the rest of the application or display an error on the authentication form.
+     * @param {func} authenticateUser The function to call to authenticate the user.
+     */
+    const onSubmit = async (authenticateUser) => {
+        const authResult = await authenticateUser();
+        if (authResult.res === "ok") {
+            proceedToAuthorize();
+        } else {
+            setError(authResult.error);
+        }
+    };
+
+    return (
+        <>
+            <AuthHeader>
+                <img className="fsbl-logo" />
+                <div className="fsbl-close">
+                    <i className="ff-close" onClick={quitApplication}></i>
+                </div>
+            </AuthHeader>
+
+            <AuthForm>
+                <div className="fsbl-login-logo" alt="Finsemble Sign In"></div>
+                <div className="fsbl-button-wrapper">
+                    {error && <ErrorMessage>
+                        {error}
+                    </ErrorMessage>}
+                    <AuthInput
+                        type="text"
+                        name="username"
+                        placeholder="Username"
+                        onChange={handleChange} 
+                    />
+                    <AuthInput
+                        type="password"
+                        name="password"
+                        placeholder="Password"
+                        onChange={handleChange} 
+                    />
+                    <AuthSubmit onClick={() => onSubmit(authenticateUser)} >
+                        Login
+                    </AuthSubmit>
+                </div>
+            </AuthForm>
+        </>
+    )
+}
+
+
+function renderAuthComponent() {
+    ReactDOM.render(<Authentication />, document.getElementById("authentication"));
+    FSBL.System.Window.getCurrent().show();
+}
+
+if (window.FSBL && FSBL.addEventListener) {
+	FSBL.addEventListener("onReady", renderAuthComponent);
```

Comment:
Since we're using the Authentication component written in this file instead of importing it from finsemble-ui, there's no better way to hide the FSBL onReady function from this file.

Reply:
We could wrap AuthForm and AuthHeader in WaitForFSBL. I suggest we do this.